docs/pages/benchmarking_and_the_dev_process.md

```diff
@@ -0,0 +1,50 @@
+# Benchmarking and the development process
+
+Using benchmarks as a development tool is all about the context those benchmarks come from and are run in. This context is important not only to help interpret the benchmarks (did performance increase or decrease, compared to what?) they also make up the history of measurement for the benchmarks we are running.
+
+Although conbench can store benchmarks from anywhere within a git tree (or even software that is outside of a git tree), we typically benchmark at specific points within a git tree for different purposes. Further, we use the git history to situate where a benchmark came from in history, what came before it, and its relationship to various points of interest in the git tree.
+
+In the most typically conbench setup we run benchmarks at different points in a repository’s git history: 
+
+- On commits to one main branch that represent the history of the software over its development and release lifecycle.
+- On PR branches, as requested (or if the benchmarks are fast enough, on each commit as it is added to a PR branch)
+
+When we compare benchmarks, we always have a _contender_ (the new code that we are considering) and a _baseline_ (the old code that were are comparing to). _Baseline_ commits will also define the history and distribution of values for a particular benchmark. This history is effectively the same benchmark measurements as they were measured through the git history (so on main, if our _baseline_ is commit HEAD-1, the history for that _baseline_ would include `HEAD-2`, `HEAD-3`, `HEAD-4`, …)
+
+## Commits to the main branch
+
+These benchmarks are the history of benchmarks that are used to track performance over time for the software we track. These are important to detect regressions that have been merged into the main branch and since we run them on every commit to the main branch, we know exactly which PR introduced a regression.
```

Comment:
This makes me realize once again that I think it's really helpful to strive for consistent terminology "benchmark" (as in the code/method) vs. "benchmark result" (data submitted to Conbench).

Reply:
Yeah, I've tried to clarify "benchmark results" when it's very clear that we're referring to those. I'm going to kick the can down the road a bit on making a kind of glossary that explained the different meanings that some folks use when they refer to "benchmarks" here. But very agreed that we need to start that consistency internally first.